scripts/lib/docker.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+
+# ensure_docker [<docker-version>]
+# 
+# Ensure that docker client is installed and daemon is running.
+ensure_docker () {
+    ensure_docker_client && ensure_docker_daemon_running
+}
+
+ensure_docker_client () {
```

Comment:
generally, I've been using the functions prefixed with `ensure_` to actually install the required tools. The problem with Docker, though, is that it's not possible to install and configure it without a) superuser privleges, b) reloading the shell and usergroups and c) doing it in a sane multi-OS way.

Which is why I've typically just put this into the top of the scripts that depend on Docker:

```
source $SCRIPTS/lib/common.sh
...
check_is_installed docker
```

which does the same as what you're doing above :)

Reply:
Haaa i see! thanks for explaining! Let's keep only `check_is_installed` then 👍 . i'll remove the docker.sh lib file

Reply:
@jaypipes i removed the `ensure_` commands. Do you think it will be useful to have `check_docker_daemon_running` or shall we keep it like this ?